Mathlib/Order/LiminfLimsup.lean

```diff
@@ -226,6 +226,30 @@ theorem IsCobounded.mono (h : f ≤ g) : f.IsCobounded r → g.IsCobounded r
   | ⟨b, hb⟩ => ⟨b, fun a ha => hb a (h ha)⟩
 #align filter.is_cobounded.mono Filter.IsCobounded.mono
 
+lemma _root_.Monotone.isCoboundedUnder_ge [Preorder X] [Preorder Y] {f : X → Y} (hf : Monotone f)
+    {u : ι → X} (F : Filter ι) [NeBot F] (bdd : F.IsBoundedUnder (· ≤ ·) u) :
+    F.IsCoboundedUnder (· ≥ ·) (f ∘ u) := by
+  apply Filter.IsBounded.isCobounded_flip
+  obtain ⟨b, hb⟩ := bdd
+  refine ⟨f b, ?_⟩
+  simp only [eventually_map, Function.comp_apply] at hb ⊢
+  filter_upwards [hb] with i hi using hf hi
+
+lemma _root_.Monotone.isCoboundedUnder_le [Preorder X] [Preorder Y] {f : X → Y} (hf : Monotone f)
+    {u : ι → X} (F : Filter ι) [NeBot F] (bdd : F.IsBoundedUnder (· ≥ ·) u) :
+    F.IsCoboundedUnder (· ≤ ·) (f ∘ u) :=
+  @Monotone.isCoboundedUnder_ge ι Xᵒᵈ Yᵒᵈ _ _ f hf.dual u F _ bdd
+
+lemma _root_.Antitone.isCoboundedUnder_le [Preorder X] [Preorder Y] {f : X → Y} (hf : Antitone f)
+    {u : ι → X} (F : Filter ι) [NeBot F] (bdd : F.IsBoundedUnder (· ≤ ·) u) :
+    F.IsCoboundedUnder (· ≤ ·) (f ∘ u) :=
+  @Monotone.isCoboundedUnder_ge ι X Yᵒᵈ _ _ f hf u F _ bdd
+
+lemma _root_.Antitone.isCoboundedUnder_ge [Preorder X] [Preorder Y] {f : X → Y} (hf : Antitone f)
+    {u : ι → X} (F : Filter ι) [NeBot F] (bdd : F.IsBoundedUnder (· ≥ ·) u) :
+    F.IsCoboundedUnder (· ≥ ·) (f ∘ u) :=
+  @Monotone.isCoboundedUnder_le ι X Yᵒᵈ _ _ f hf u F _ bdd
```

Comment:
I have some ideas about how to make these a bit stronger, but it would be a bit long to explain. Is it fine if I just open a PR so you can look at it?

Reply:
Actually explaining the math idea is not hard: the underlying result is really that, if `F.IsBoundedUnder (· ≤ ·) u)` and `u` is monotone, then `F.IsBoundedUnder (· ≤ ·) (f ∘ u)`, which implies your result. The thing that would be long to explain is how I suggest setting up the API :sweat_smile:

Reply:
That would be great, of course! Thanks! (For me these were just a way to get unstuck in the proofs of the lemmas mentioned in the PR description, which in turn were mainly a way to get unstuck elsewhere...)

Reply:
I've opened that PR as #6287. For example, if `hf : Monotone f` and `bdd : F.IsBoundedUnder (· ≤ ·) u` you have `(hf.isBoundedUnder_le bdd).isCoboundedUnder_ge : F.IsCoboundedUnder (· ≥ ·) (f ∘ u)`. If that is too long you can keep the `Cobounded` version too of course, and prove it that way.